sonic-net/sonic-buildimage
# [image_config]: Enable Receive Packet Steering (RPS)

* This patch enables Receive Packet Steering (RPS) which helps to distribute softirq processing for CPU bound packets to all 
    available cores. This will help prevent kernel packet drops when there are bouts of intense CPU bound packets.

    Ref: https://lwn.net/Articles/362339/ 
           https://docs.kernel.org/networking/scaling.html



#### Why I did it
To enable CPU to handle CPU bound packets more efficiently
##### Work item tracking
- Microsoft ADO **27251849**:

#### How I did it
By using a systemd service that enables RPS for all front-panel ports on a sonic switch
#### How to verify it
Verify that "/sys/class/net/Ethernet<x>/queues/rx-0/rps_cpus" is set to all available CPU cores


#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)



#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
